packages/native/src/useUnhandledLinking.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+  getStateFromPath,
+  NavigationProp,
+  ResultState,
+  useNavigation,
+} from '@react-navigation/core';
+import React from 'react';
+import { Platform } from 'react-native';
+import useLatestCallback from 'use-latest-callback';
+
+import { extractPathFromURL } from './extractPathFromURL';
+import { LinkingContext } from './LinkingContext';
+
+function extractNavigatorSpecifcState(
+  navigation: NavigationProp<ReactNavigation.RootParamList>,
+  rootState: ResultState
+) {
+  let state: typeof rootState | undefined = rootState;
+
+  const parent = navigation.getParent();
+  if (parent) {
+    // Then, we consider a portion of the state.
+    const parentState = parent.getState();
+    const outerRouteName = parentState.routes[parentState.index].name;
+    while (state && state.routes[state.index ?? 0].name !== outerRouteName) {
+      state = state.routes[state.index ?? 0].state;
+    }
+    if (!state) {
+      return;
+    }
+    const innerState = state.routes[state.index ?? 0].state;
+    if (!innerState) {
+      return;
+    }
+    state = innerState;
+  }
+  return state;
+}
+
+export function useUnhandledLinking() {
+  const navigation = useNavigation();
+  const linking = React.useContext(LinkingContext);
+
+  const { options, lastUnhandledLinking } = linking;
+
+  /*
+   * Function to handle last unhandled URL. This function has to be called when the conditional
+   * rendering of the navigator is about to happen e.g. in the `onPress` of a log in button.
+   */
+  const handleLastLinking = () => {
+    if (options == null || lastUnhandledLinking?.current == null) {
```

Comment:
we still handle deep links when `options` isn't defined.

```suggestion
    if (lastUnhandledLinking?.current == null) {
```